pkg/controllers/management/drivers/kontainerdriver/kontainerdriver.go

```diff
@@ -107,12 +107,21 @@ func (l *Lifecycle) driverExists(obj *v3.KontainerDriver) bool {
 
 func (l *Lifecycle) download(obj *v3.KontainerDriver) (*v3.KontainerDriver, error) {
 	driver := drivers.NewKontainerDriver(obj.Spec.BuiltIn, obj.Name, obj.Spec.URL, obj.Spec.Checksum)
-	err := driver.Stage()
-	if err != nil {
-		return nil, err
-	}
 
-	v3.KontainerDriverConditionDownloaded.True(obj)
+	newObj, err := v3.KontainerDriverConditionDownloaded.Once(obj, func() (runtime.Object, error) {
+		// update status
+		obj, err := l.kontainerDrivers.Update(obj)
```

Comment:
the object condition `KontainerDriverConditionInstalled` will be updated to `Unknown` too. Maybe we should set Installed condition to Unkonwn after download is success.

Reply:
I do not think it is necessary because we update nothing for this condition, but I added it before the condition changed, maybe it is better.

Reply:
Please move this `update` out of `Installed.Once(...)`, it is very confusing.